oauth2/custom-grant/src/main/java/org/wso2/sample/identity/oauth2/grant/mobile/MobileGrant.java

```diff
@@ -64,8 +66,17 @@ public boolean validateGrant(OAuthTokenReqMessageContext oAuthTokenReqMessageCon
 
             if(authStatus) {
                 // if valid set authorized mobile number as grant user
+                String tenantDomain = MultitenantUtils.getTenantDomain(mobileNumber);
+                String tenantAwareUsername = MultitenantUtils.getTenantAwareUsername(mobileNumber);
                 AuthenticatedUser mobileUser = new AuthenticatedUser();
-                mobileUser.setUserName(mobileNumber);
+                mobileUser.setUserName(tenantAwareUsername);
+                mobileUser.setTenantDomain(tenantDomain);
+                /*
+                   Set the federated IdP name as "LOCAL" to avoid constraint violation errors. If a federated user is
+                   involved with this custom grant then, the federated IdP name and the other claims should be fetched
+                   accordingly.
+                 */
+                mobileUser.setFederatedIdPName(FrameworkConstants.LOCAL_IDP_NAME);
```

Comment:
Let's suggest how to handle the user locally existing scenario as well. It seems good to use `AuthenticatedUser.createLocalAuthenticatedUserFromSubjectIdentifier(username)` and `AuthenticatedUser.createFederateAuthenticatedUserFromSubjectIdentifier(username)` to reduce the complexity.

Reply:
This issue is fixed with https://github.com/wso2/samples-is/pull/321/commits/3ae73a2b868d2d3d9f5abf3c1a41ff0d0ccab129.